src/PlaylistOnArray.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * {@code playlist} of {@code Track} represented on an array.
+ *
+ * @convention {@code playlist} is an array of size 100 of {@code Track}. All
+ *             spots in the array should be filled with the default value of
+ *             {@code track}. There should be no spots with default value
+ *             between spots that have been filled in.
+ * @correspondence {@code playlist} corresponds to the list of {@code Track}.
+ *                 {@code this[i]} represents the {@code Track} at position
+ *                 {@code i}.
+ */
+public class PlaylistOnArray extends PlaylistSecondary {
+
+    class Track {
```

Comment:
This is great! You might look into Java records, which simplify the code here a little bit.